src/bot/lib/query.js

```diff
@@ -30,18 +30,19 @@ export function findSkillByName(token, search) {
 }
 
 // Find the best match for the given search term, and complain if necessary.
-export function findSkillAndHandleErrors(token, search) {
+export function findItemAndHandleErrors(token, search, queryFunction) {
```

Comment:
The only problem left is that this refactoring removes some of the error catching. I have a stand in 5 minutes but I'll follow up after on how we can address that!